lib/ansible/modules/windows/win_chocolatey.ps1

```diff
@@ -69,8 +72,19 @@ Function Chocolatey-Install-Upgrade
     if ($ChocoAlreadyInstalled -eq $null)
     {
 
-        # We need to install chocolatey
-        $install_output = (new-object net.webclient).DownloadString("https://chocolatey.org/install.ps1") | powershell -
+        #We need to install chocolatey
+        $wc = New-Object System.Net.WebClient;
+        if ($proxy_url)
+        {
+            #We need to configure proxy
+            $wp = New-Object System.Net.WebProxy($proxy_url, $true);
```

Comment:
As well as setting the initial proxy so the module can download the install script, the install script requires the following values so it can download it's dependencies when using a proxy.

```
$env:chocolateyProxyLocation = 'https://local/proxy/server'
$env:chocolateyProxyUser = 'username'
$env:chocolateyProxyPassword = 'password'
```

Are you able to set these as well if required.

Reply:
Added. it was working without those lines last time I tried.